secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/vrp/DomesticVrpConsentsApiController.java

```diff
@@ -0,0 +1,202 @@
+
+import com.forgerock.sapi.gateway.ob.uk.common.datamodel.converter.v4.vrp.FRDomesticVRPConsentConverters;
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBErrorResponseException;
+import com.forgerock.sapi.gateway.ob.uk.rs.obie.api.payment.v4_0_0.vrp.DomesticVrpConsentsApi;
+import com.forgerock.sapi.gateway.ob.uk.rs.server.api.obie.payment.factories.v4_0_0.OBDomesticVRPConsentResponseFactory;
+import com.forgerock.sapi.gateway.ob.uk.rs.server.service.balance.FundsAvailabilityService;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.OBValidationService;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.v4.payment.consent.OBVRPFundsConfirmationRequestValidator.VRPFundsConfirmationValidationContext;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.payment.vrp.DomesticVRPConsentStoreClient;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.vrp.v3_1_10.CreateDomesticVRPConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.vrp.v3_1_10.DomesticVRPConsent;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPConsentRequest;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPConsentResponse;
+import uk.org.openbanking.datamodel.v4.vrp.OBPAFundsAvailableResult1;
+import uk.org.openbanking.datamodel.v4.vrp.OBPAFundsAvailableResult1FundsAvailable;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationRequest;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationResponse;
+import uk.org.openbanking.datamodel.v4.vrp.OBVRPFundsConfirmationResponseData;
+
+@jakarta.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+@Controller("DomesticVrpConsentApiV4.0.0")
+@Slf4j
+public class DomesticVrpConsentsApiController implements DomesticVrpConsentsApi {
+
+    private final FundsAvailabilityService fundsAvailabilityService;
+
+    private final DomesticVRPConsentStoreClient consentStoreClient;
+
+    private final OBValidationService<OBDomesticVRPConsentRequest> vrpConsentValidator;
+
+    private final OBValidationService<VRPFundsConfirmationValidationContext> vrpFundsConfirmationValidator;
+
+    private final OBDomesticVRPConsentResponseFactory responseFactory;
+
+    public DomesticVrpConsentsApiController(FundsAvailabilityService fundsAvailabilityService,
+                                            @Qualifier("v4.0.0RestDomesticVRPConsentStoreClient")DomesticVRPConsentStoreClient consentStoreClient,
+                                            OBValidationService<OBDomesticVRPConsentRequest> vrpConsentValidator,
+                                            OBValidationService<VRPFundsConfirmationValidationContext> vrpFundsConfirmationValidator,
+                                            OBDomesticVRPConsentResponseFactory responseFactory) {
+        this.fundsAvailabilityService = fundsAvailabilityService;
```

Comment:
Should we not be null-checking these values here?

Reply:
Added some null warnings just in case but these controllers are copies from v3.1.10 implementation adapted to use the v4.0.0 objects.